kfs-core/src/main/java/edu/arizona/kfs/tax/document/IncomeTypeHandler.java

```diff
@@ -0,0 +1,576 @@
+			LOG.debug("validateIncomeType() retval: " + retval);
+		}
+
+		return retval;
+	}
+
+	public CurrencyFormatter getCurrencyFormatter() {
+		return currencyFormatter;
+	}
+
+	public T1 getNewIncomeType() {
+		if (newIncomeType == null) {
+			newIncomeType = getInitializedDocumentIncomeType();
+		}
+
+		return newIncomeType;
+	}
+
+	private T1 getInitializedDocumentIncomeType() {
+		T1 retval = null;
+
+		try {
+			// use the class passed in the constructor to create a new instance
+			retval = incomeTypeClass.newInstance();
+
+			// initialize amount to 0
+			retval.setAmount(new KualiDecimal(0));
+
+			// set the document identifier from the document
+			retval.setDocumentIdentifier(incomeTypeDocumentContainer.getDocumentIdentifier());
+		}
+
+		catch (Exception ex) {
+			LOG.error(ex);
+		}
+
+		return retval;
+	}
+
+	public void setNewIncomeType(T1 newIncomeType) {
+		this.newIncomeType = newIncomeType;
+	}
+
+	/**
+	 * This method loads IncomeTypes into a map keyed by box number
+	 * 
+	 * @return
+	 */
+	private Map<String, IncomeType> getIncomeTypeMap() {
+		Map<String, IncomeType> retval = new HashMap<String, IncomeType>();
+
+		// Load the IncomeTypes from the database
+		List<IncomeType> incomeTypes = (List<IncomeType>) getBusinessObjectService().findAll(IncomeType.class);
+
+		if (incomeTypes != null) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("found " + incomeTypes.size() + " income types");
+			}
+
+			// create IncomeType map by 1099 box
+			for (IncomeType ic : incomeTypes) {
+				if (StringUtils.isNotBlank(ic.getIncomeTypeBox())) {
+					if (LOG.isDebugEnabled()) {
+						LOG.debug("IncomeTypeCode: " + ic.getIncomeTypeCode() + ", box: " + ic.getIncomeTypeBox());
+					}
+
+					retval.put(ic.getIncomeTypeBox(), ic);
+				}
+			}
+		}
+
+		return retval;
+	}
+
+	/**
+	 * This method populates the 1099 IncomeTypes from the document accounting lines
+	 * 
+	 * @param accountingLines
+	 * @param useTaxItems
+	 */
+	public void populateIncomeTypes(List<? extends AccountingLine> accountingLines) {
+		VendorHeader vendorHeader = incomeTypeDocumentContainer.getVendorHeader();
+
+		// if we found the vendor
+		if (vendorHeader != null) {
+			// loop over the accounting lines associated with the documents and summarize
+			// valid 1099 accounting lines by income type (box on the 1099 form)
+			for (AccountingLine acctline : accountingLines) {
+				// get the 1099 box for this account
+				String box = get1099Box(vendorHeader, acctline);
+
+				if (StringUtils.isNotBlank(box)) {
+					// See if a DocumentIncomeType associated with this 1099 box
+					// exists in out map
+					T1 dit = getDocumentIncomeType(acctline, box);
+
+					if (dit != null) {
+						// add the amount from the current accounting line to the DocumentIncomeType
+						dit.setAmount(dit.getAmount().add(acctline.getAmount()));
+					}
+				}
+			}
+		}
+	}
+
+	private String get1099Box(VendorHeader vendorHeader, AccountingLine acctline) {

```

Comment:
Each of these get1099Box methods are only used in 1 place. A comment has been added to both each, indicating what methods use them.